starcheck/utils.py

```diff
@@ -17,119 +29,113 @@
-            raise
-    return wrapper
+def date2secs(val):
+    """Convert date to seconds since 1998.0"""
+    out = cxotime.date2secs(val)
+    # if isinstance(out, (np.number, np.ndarray)):
+    #     out = out.tolist()
+    return out
+
+
+def secs2date(val):
+    """Convert date to seconds since 1998.0"""
+    out = cxotime.secs2date(val)
+    # if isinstance(out, (np.number, np.ndarray)):
+    #    out = out.tolist()
+    return out
 
 
-# Borrowed from https://stackoverflow.com/a/33160507
-def de_bytestr(data):
-    if isinstance(data, bytes):
-        return data.decode()
-    if isinstance(data, dict):
-        return dict(map(de_bytestr, data.items()))
-    if isinstance(data, tuple):
-        return tuple(map(de_bytestr, data))
-    if isinstance(data, list):
-        return list(map(de_bytestr, data))
-    if isinstance(data, set):
-        return set(map(de_bytestr, data))
-    return data
+def date2time(val):
+    out = cxotime.date2secs(val)
+    if isinstance(out, (np.number, np.ndarray)):
+        out = out.tolist()
+    return out
 
 
-date2time = python_from_perl(date2secs)
-time2date = python_from_perl(secs2date)
+def time2date(val):
+    out = cxotime.secs2date(val)
+    if isinstance(out, (np.number, np.ndarray)):
+        out = out.tolist()
+    return out
 
 
-@python_from_perl
-def ccd_temp_wrapper(kwargs):
+def ccd_temp_wrapper(**kwargs):
     return get_ccd_temps(**kwargs)
 
 
-@python_from_perl
-def plot_cat_wrapper(kwargs):
+def plot_cat_wrapper(**kwargs):
     return make_plots_for_obsid(**kwargs)
 
 
-@python_from_perl
 def starcheck_version():
     return version
 
 
-@python_from_perl
 def get_chandra_models_version():
     return proseco_char.chandra_models_version
 
 
-@python_from_perl
 def set_kadi_scenario_default():
     # For kadi commands v2 running on HEAD set the default scenario to flight.
     # This is aimed at running in production where the commands archive is
     # updated hourly. In this case no network resources are used.
     if test_helper.on_head_network():
-        os.environ.setdefault('KADI_SCENARIO', 'flight')
+        os.environ.setdefault("KADI_SCENARIO", "flight")
 
 
-@python_from_perl
 def get_cheta_source():
     sources = starcheck.calc_ccd_temps.fetch.data_source.sources()
-    if len(sources) == 1 and sources[0] == 'cxc':
-        return 'cxc'
+    if len(sources) == 1 and sources[0] == "cxc":
+        return "cxc"
     else:
         return str(sources)
 
 
-@python_from_perl
 def get_kadi_scenario():
-    return os.getenv('KADI_SCENARIO', default="None")
+    return os.getenv("KADI_SCENARIO", default="None")
 
 
-@python_from_perl
 def get_data_dir():
-    sc_data = os.path.join(os.path.dirname(starcheck.__file__), 'data')
+    sc_data = os.path.join(os.path.dirname(starcheck.__file__), "data")
     return sc_data if os.path.exists(sc_data) else ""
 
 
-@python_from_perl
-def _make_pcad_attitude_check_report(kwargs):
+def _make_pcad_attitude_check_report(**kwargs):
```

Comment:
This is an API change (which is fine... they are just a little hard to see in the changes in here so I'm commenting now for myself).